docs/src/main/asciidoc/azure-app-config-client.adoc

```diff
@@ -0,0 +1,161 @@
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 11+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven {maven-version}
+
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+
+== Stand up a Azure Config Server
+
+To stand up the Azure Config Server required for this guide, please follow the instructions outlined https://docs.microsoft.com/en-us/azure/azure-app-configuration/quickstart-java-spring-app[here] till 'Create a Spring Boot app'.
+The end result of that process is a running Azure App Config Server that will provide the `Hello` value for a configuration property named `/application/config.message`.
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=azure-app-config-quickstart \
+    -DclassName="org.acme.azure.app.config.client.GreetingResource" \
+    -Dpath="/greeting" \
+    -Dextensions="azure-app-config-client"
+cd azure-app-config-quickstart
+----
+
+This command generates a Maven project with a REST endpoint and imports the `azure-app-config-client` extension.
+
+If you already have your Quarkus project configured, you can add the `azure-app-config-client` extension
+to your project by running the following command in your project base directory:
+
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="azure-app-config-client"
+----
+
+This will add the following to your `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-azure-app-config-client</artifactId>
+</dependency>
+----
+
+== GreetingController
+
+The Quarkus Maven plugin automatically generated a `GreetingResource` JAX-RS resource in the
+`src/main/java/org/acme/azure/app/config/client/GreetingResource.java` file that looks like:
+
+[source,java]
+----
+package org.acme.azure.app.config.client;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+}
+----
+
+As we want to use configuration properties obtained from the Config Server, we will update the `GreetingResource` to inject the `message` property. The updated code will look like this:
+
+[source,java]
+----
+package org.acme.azure.app.config.client;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@Path("/hello")
+public class GreetingResource {
+
+    @ConfigProperty(name = "application/config.message", defaultValue="hello default")
```

Comment:
`"application/config.message"` is something we don't want. All configuration properties should only use `.` as a delimiter